pkg/agent/agent.go

```diff
@@ -1197,11 +1206,37 @@ func (i *Initializer) initVMLocalConfig(nodeName string) error {
 		return err
 	}
 
+	// TODO: Handle for multiple interfaces
+	var uplinkInterface *net.Interface
+	foundNetDevice := false
+	for _, addr := range en.Spec.Interfaces[0].IPs {
+		var ipFilter *utilip.DualStackIPs
+		var err error
+		epIP := net.ParseIP(addr)
+		if epIP.To4() != nil {
+			ipFilter = &utilip.DualStackIPs{IPv4: epIP}
+		} else {
+			ipFilter = &utilip.DualStackIPs{IPv6: epIP}
+		}
+		_, _, uplinkInterface, err = util.GetIPNetDeviceFromIP(ipFilter, sets.NewString())
+		if err != nil {
+			klog.InfoS("Failed to get net device for", "IP", addr)
+		} else {
+			foundNetDevice = true
+			klog.V(2).InfoS("Using the ExternalNode", "InterfaceName", uplinkInterface.Name)
+			break
+		}
+	}
+	if !foundNetDevice {
+		return fmt.Errorf("failed to get net device for interface %s", uplinkInterface.Name)
```

Comment:
uplinkInterface is possible nil if `foundNetDevice==false`. You can print the IPs in the interface if you think that is important.

Reply:
we do log the IPs in the line for which we were not able to find on the device at L1223, so it should be fine.

Reply:
My point is `uplinkInterface` is possible nil, and it may crash when calling `uplinkInterface.Name`

Reply:
good catch. Updated it.